source/common/network/listener_filter_buffer_impl.cc

```diff
@@ -102,7 +102,9 @@ absl::Status ListenerFilterBufferImpl::onFileEvent(uint32_t events) {
   ASSERT(events == Event::FileReadyType::Read);
 
   auto state = peekFromSocket();
-  if (state == PeekState::Done) {
+  if (state == PeekState::Done && !on_data_cb_disabled_) {
+    // buffer_size_ will be set to 1 if the first listener filter in
+    // filter chain has maxReadBytes() of 0. Bypass onData callback of it.
```

Comment:
> buffer_size_ will be set to 1

why we need to do this?

Reply:
Because if we set buffer size to 0 for listener filters with maxReadBytes() of 0, in theory we should only initialize Event::FileReadyType::Closed. But there are some limitations on different platforms (Linux, macOS, Windows), e.g. on macOS the connection doesn't get early close notification and instead gets the close after reading the FIN, which means Event::FileReadyType::Read is required

Reply:
Ok, please document it clear. Or the future reader will be confused.